rubix/galaxy/input_handler/pynbody.py

```diff
@@ -84,6 +88,35 @@ def load_data(self):
                     getattr(self.sim, cls), fields[cls], units[cls], cls
                 )
 
+        # Combine HI and OxMassFrac into a two-column metals field for gas
+        hi_data = self.load_particle_data(
+            getattr(self.sim, "gas"),
+            {"HI": "HI"},
+            {"HI": u.dimensionless_unscaled},
+            "gas",
+        )
+        ox_data = self.load_particle_data(
+            getattr(self.sim, "gas"),
+            {"OxMassFrac": "OxMassFrac"},
+            {"OxMassFrac": u.dimensionless_unscaled},
+            "gas",
+        )
+        # fe_data = self.load_particle_data(getattr(self.sim, "gas"), {"FeMassFrac": "FeMassFrac"}, {"FeMassFrac": u.dimensionless_unscaled}, "gas")
+        # self.data["gas"]["metals"] = np.column_stack((hi_data["HI"], ox_data["OxMassFrac"]))
+        # Create a metals array with 10 columns, filled with zeros initially
+        n_particles = hi_data["HI"].shape[0]
+        metals = np.zeros((n_particles, 10), dtype=hi_data["HI"].dtype)
+
+        # Place HI values at column 0 and OxMassFrac (O) at column 4 (that it is storred in the same way as IllustrisTNG)
+        metals[:, 0] = hi_data["HI"]
+        metals[:, 4] = ox_data["OxMassFrac"]
+
+        self.data["gas"]["metals"] = metals
+        self.logger.info("Metals assigned to gas particles.")
+        self.logger.info("Metals shape is: %s", self.data["gas"]["metals"].shape)
+
+        self.data["stars"]["age"] = 14.14019767 * u.Gyr - self.data["stars"]["age"]
```

Comment:
why do we have 14.14 Gyr? shouldn't the universe be 13.8 Gyr?

Reply:
probably that should be connected to the cosmology we specify in the yaml file...
see here:
https://github.com/AstroAI-Lab/rubix/blob/dbb44870257411aa223acdb80b578dfffe42e305/rubix/cosmology/base.py#L234